protocol/persistence.go

```diff
@@ -1104,3 +1109,107 @@ func (db *sqlitePersistence) DeleteSoftRemovedBookmarks(threshold uint64) error
 	_, err = tx.Exec(`DELETE from bookmarks WHERE removed = 1 AND deleted_at < ?`, threshold)
 	return err
 }
+
+func (db *sqlitePersistence) InsertWalletConnectSession(peerID string, connectorInfo string) error {
+	tx, err := db.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		_ = tx.Rollback()
+	}()
+
+	sessionInsertPreparedStatement, err := tx.Prepare("INSERT OR REPLACE INTO wallet_connect_sessions(peer_id, connector_info) VALUES(?, ?)")
+	if err != nil {
+		return err
+	}
+	_, err = sessionInsertPreparedStatement.Exec(peerID, connectorInfo)
+	sessionInsertPreparedStatement.Close()
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func (db *sqlitePersistence) GetWalletConnectSession() (Session, error) {
+	tx, err := db.db.Begin()
+
+	seshObject := Session{
+		PeerID:        "",
+		ConnectorInfo: "",
+	}
+	if err != nil {
+		return seshObject, err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		_ = tx.Rollback()
+	}()
+
+	rows, err := tx.Query("SELECT * FROM wallet_connect_sessions")
```

Comment:
What's the expected behavior in case there's more than one record in wallet_connect_sessions? use always the last row returned from the query?

Reply:
currently there will only be 1 session so only 1 row, multiple sessions are still a WIP